presto-main/src/main/java/io/prestosql/sql/planner/optimizations/AddExchanges.java

```diff
@@ -256,8 +257,12 @@ public PlanWithProperties visitGroupId(GroupIdNode node, PreferredProperties pre
         @Override
         public PlanWithProperties visitMarkDistinct(MarkDistinctNode node, PreferredProperties preferredProperties)
         {
-            PreferredProperties preferredChildProperties = PreferredProperties.partitionedWithLocal(ImmutableSet.copyOf(node.getDistinctSymbols()), grouped(node.getDistinctSymbols()))
-                    .mergeWithParent(preferredProperties);
+            PreferredProperties preferredChildProperties = PreferredProperties.partitionedWithLocal(ImmutableSet.copyOf(node.getDistinctSymbols()), grouped(node.getDistinctSymbols()));
+
+            if (!isForcePartitioningMarkDistinctInput(session)) {
+                preferredChildProperties = preferredChildProperties.mergeWithParent(preferredProperties);
```

Comment:
what does `preferredChildProperties.mergeWithParent(preferredProperties);` change here?

In both cases (when `isForcePartitioningMarkDistinctInput` is enabled or disabled) we still use preferred properties:
```
PreferredProperties.partitionedWithLocal(ImmutableSet.copyOf(node.getDistinctSymbols()), grouped(node.getDistinctSymbols()));
```
where we want partitioning by `node.getDistinctSymbols()`

Reply:
But when we merge with it's parent the preferred partitioned will be based on the common symbols between both properties. (which leads to skew as in #574)

Reply:
I'm not sure it should be operator specific. Rather it might need to be approached holistically, see also https://github.com/prestodb/presto/issues/11025

Reply:
I agree. We should have a flag that enables/disables this behavior across the board. Basically, it allows to trade off network transfers for better distribution for various operators.